test/CMakeLists.txt

```diff
@@ -113,6 +113,8 @@ endif()
 if(WITH_GTEST)
     include(FetchContent)
 
+    find_package(GTest)
```

Comment:
We need to find a system ``GTest`` only if the cmake target ``GTest::GTest`` is not defined yet (condition on line 130).

Reply:
When built alone, nothing defines target `GTest::GTest` before that line... I'm still testing if using system version of `googletest` breaks memory sanitizer.

Reply:
It may be defined if using zlib-ng as a subproject.

Reply:
> It may be defined if using zlib-ng as a subproject.

True, but it's chicken and egg kind of problem, it can fail and thus we would need to check for existence of the target twice...

Reply:
See PR #1277

Reply:
If the target doesn't exist, call find_package, like we did in minizip-ng: https://github.com/zlib-ng/minizip-ng/issues/654. Then if find_package fails, we fetchcontent instead?

Reply:
> True, but it's chicken and egg kind of problem, it can fail and thus we would need to check for existence of the target twice...

Yes. We need twice target check.
Suggestion:

Replace:

```cmake
find_package(GTest)
```

with:

```cmake
if(NOT TARGET GTest::GTest)
    find_package(GTest)
endif()
```

Reply:
Testing for existence of target `GTest::GTest` has been deprecated since cmake 3.20... Preferred way is to use `find_package` which doesn't try to find the dependency twice as the result is cached...

Reply:
Sorry, I'd completely forgot that `find_package` got substantially smarter!

I got caught out with that before.

Reply:
We need to support cmake versions down to 3.5, which limits what parameters we can use for some cmake functions.

Reply:
> Testing for existence of target `GTest::GTest` has been deprecated since cmake 3.20... Preferred way is to use `find_package` which doesn't try to find the dependency twice as the result is cached...

This is true only for twice find_package call.
If in the parent project GTest is included by a different way (not by find_package), then there may be a library conflict.

We can replace all of ``GTest::GTest`` with ``GTest::gtest`` (https://cmake.org/cmake/help/latest/module/FindGTest.html), but anyway we need add condition before find_package.

Reply:
As using system copy of googletest breaks with memory sanitizer enabled, I've moved the `find_package()` call later in the file... For now we will do the target test twice to make it work with both standalone build and as a subproject.